src/libraries/Microsoft.Extensions.Configuration/src/InternalConfigurationRootExtensions.cs

```diff
@@ -27,6 +27,7 @@ internal static IEnumerable<IConfigurationSection> GetChildrenImplementation(thi
                 .Aggregate(Enumerable.Empty<string>(),
                     (seed, source) => source.GetChildKeys(seed, path))
                 .Distinct(StringComparer.OrdinalIgnoreCase)
+                .OrderBy(key => key)
```

Comment:
Should this use the same comparer for configuration keys that the code you deleted did for sorting?